src/help/help_impl.cpp

```diff
@@ -1332,181 +1334,381 @@ section *find_section(section &sec, const std::string &id)
+However, no paragraph break should be used when [text] is followed by something else.
+It is possible to have empty text spans in some cases, for example given a run of more than 2 newlines,
+or a character entity directly followed by a paragraph break.
+
+*/
+static config parse_entity(std::string::const_iterator& beg, std::string::const_iterator end)
+{
+	config entity;
+	std::stringstream s;
+	enum { UNKNOWN, NAMED, HEX, DECIMAL } type = UNKNOWN;
+	assert(*beg == '&');
+	++beg;
+	for(; beg != end && *beg != ';'; ++beg) {
+		switch(type) {
+		case UNKNOWN:
+			if(*beg == '#') {
+				type = DECIMAL;
+			} else if(isalnum(*beg) || *beg == '_') {
+				type = NAMED;
+				s << *beg;
+			} else {
+				throw parse_error("TODO");
+			}
+			break;
+		case NAMED:
+			if(!isalnum(*beg)) {
+				throw parse_error("TODO");
+			}
+			s << *beg;
+			break;
+		case DECIMAL:
+			if(*beg == 'x') {
+				type = HEX;
+			} else if(isdigit(*beg)) {
+				s << *beg;
+			} else {
+				throw parse_error("TODO");
 			}
-			last_char_escape = false;
+			break;
+		case HEX:
+			if(isxdigit(*beg)) {
+				s << *beg;
+			} else {
+				throw parse_error("TODO");
+			}
+			break;
+		}
+	}
+	if(type == NAMED) {
+		std::string name = s.str();
+		entity["name"] = name;
+		if(name == "lt") {
+			entity["code_point"] = '<';
+		} else if(name == "gt") {
+			entity["code_point"] = '>';
+		} else if(name == "apos") {
+			entity["code_point"] = '\'';
+		} else if(name == "quot") {
+			entity["code_point"] = '"';
+		} else if(name == "amp") {
+			entity["code_point"] = '&';
+		}
+	} else {
+		s.seekg(0);
+		if(type == HEX) {
+			s >> std::hex;
 		}
+		int n;
+		s >> n;
+		entity["code_point"] = n;
+	}
+	return entity;
+}
+
+static char parse_escape(std::string::const_iterator& beg, std::string::const_iterator end)
+{
+	assert(*beg == '\\');
+	// An escape at the end of stream is just treated as a literal.
+	// Otherwise, take the next character as a literal and be done with it.
+	if((beg + 1) != end) {
+		++beg;
 	}
-	if (state == ELEMENT_NAME) {
-		std::stringstream msg;
-		msg << "Element '" << ss.str() << "' continues through end of string.";
-		throw parse_error(msg.str());
+	return *beg;
+}
+
+static config parse_text_until(std::string::const_iterator& beg, std::string::const_iterator end, char close)
+{
+	// In practice, close will be one of < ' "
+	// Parsing will go until either close or eos, and will emit one or more text and character_entity tags.
+	// However, recognized character entities will be collapsed into the text tags.
+	std::ostringstream s;
+	bool saw_newline = false;
+	config res;
+	for(; beg != end && *beg != close; ++beg) {
+		if(*beg == '&') {
+			auto entity = parse_entity(beg, end);
+			if(beg == end) {
+				throw parse_error("unexpected eos after entity");
+			}
+			if(entity.has_attribute("code_point")) {
+				std::wstring_convert<std::codecvt_utf8<char32_t>, char32_t> conv;
```

Comment:
Apparently `std::wstring_convert` and `std::codecvt_utf8` are deprecated in C++17. What's the desired behavior here?

Reply:
@CelticMinstrel

Reply:
Well, this is doing exactly what it's intended to do, and I don't see anything noted as a replacement for the functionality. It converts an integer unicode code point into its UTF-8 representation.

Since it's not removed until C++26, I think we should leave it as-is for this PR… unless you know what it can be replaced with?

Reply:
But where do we use that functionality.

Reply:
What are you asking?

Reply:
Sorry, was unclear. I meant where is `code_point` being set? Where are we saving unicode codepoints instead of just the character?

Reply:
This code is for replacing an XML-style character entity of the form `&#nnn;` or `&#xnnnn` with the Unicode character that that represents. The `"code_point"` attribute is set in the function that parses the entity.